pkg/auth/hook_handler.go

```diff
@@ -0,0 +1,55 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/skygeario/skygear-server/pkg/auth/response"
+
+	"github.com/skygeario/skygear-server/pkg/core/db"
+	"github.com/skygeario/skygear-server/pkg/core/handler"
+)
+
+type HookHandler interface {
+	DecodeRequest(request *http.Request) (handler.RequestPayload, error)
+	WithTx() bool
+
+	ExecBeforeHooks(payload interface{}, user *response.User) error
+	HandleRequest(payload interface{}, user *response.User) (interface{}, error)
+	ExecAfterHooks(payload interface{}, resp interface{}, user response.User) error
+}
+
+type hookExecutor struct {
+	handler HookHandler
+}
+
+func HookHandlerToAPIHandler(h HookHandler, txContext db.TxContext) http.Handler {
```

Comment:
Name should be `HookHandlerToHandler`? or it should return the skygear api handler.